From 7f0f7eb77291c180218ed3cee0c297c58fa00555 Mon Sep 17 00:00:00 2001
Message-Id: <7f0f7eb77291c180218ed3cee0c297c58fa00555.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Tomas Elf <tomas.elf@intel.com>
Date: Wed, 21 Jan 2015 13:48:25 +0000
Subject: [PATCH 52/61] FOR_UPSTREAM [VPG]: drm/i915: Updated TDR to work with
 the scheduler

Added TDR-scheduler integration point to i915_handle_error

Change-Id: I1b1c59da17fb638eaeb57657468a2d599da852d4
For: VIZ-1587
For: VIZ-4741
For: GMIN-3638
Signed-off-by: Tomas Elf <tomas.elf@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 2eaff65..038b88e2 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -3069,6 +3069,15 @@ void i915_handle_error(struct drm_device *dev, struct intel_ring_hangcheck *hc,
 	va_list args;
 	char error_msg[80];
 
+	if (hc) {
+		struct intel_engine_cs *ring = &dev_priv->ring[hc->ringid];
+
+		if (!i915_scheduler_is_ring_flying(ring)) {
+			DRM_DEBUG_TDR("False hang detection on %s. Aborting!\n", ring->name);
+			return;
+		}
+	}
+
 	va_start(args, fmt);
 	vscnprintf(error_msg, sizeof(error_msg), fmt, args);
 	va_end(args);
-- 
1.7.9.5

